karapace/karapace.py

```diff
@@ -21,16 +21,7 @@ def __init__(self, config: Config) -> None:
         self.kafka_timeout = 10
         self.route("/", callback=self.root_get, method="GET")
         self.log = logging.getLogger("Karapace")
-        self.app.on_startup.append(self.create_http_client)
         self.log.info("Karapace initialized")
-        self.app.on_shutdown.append(self.close_by_app)
```

Comment:
This didn't really work. I _think_ the eventloop is closed when shutdown is called, if that is correct then the execution failed at some point in time, when running a task, and not everything was closed.

This PR consolidates all the cleanup in the base class, and uses `on_cleanup` instead, which works as expected.